### PR TITLE
Feature notes crud

### PR DESCRIPTION
**Pull request by OmKakatkar**
# Features

## Additions
* CRUD operations on notes
* A modal for updating notes

## Notes
* All style changes will be done in a future PR

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -7,20 +7,23 @@ import SignUp from "./frontend/pages/Auth/SignUp/SignUp";
 import MockAPI from "./frontend/mock/MockAPI";
 import UserProfile from "./frontend/pages/UserProfile/UserProfile";
 import { ToastContainer } from "react-toastify";
+import NewNote from "./frontend/pages/NewNote/NewNote";
```
Comment (praveenreddy1798) [inline src/App.js:10]: use barrel imports

File: src/App.js [hunk 2]
```diff
@@ -7,20 +7,23 @@ import SignUp from "./frontend/pages/Auth/SignUp/SignUp";
...
+import Notes from "./frontend/pages/Notes/Notes";
```
Comment (praveenreddy1798) [inline src/App.js:11]: use Barrel Imports

File: src/frontend/components/CreateNote/CreateNote.jsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+function CreateNote() {
+	const { user } = useAuth();
+	const { dispatch } = useNotes();
+	const navigate = useNavigate();
+
+	const [noteData, setNoteData] = useState({
+		title: "",
+		description: "",
+		noteColor: "#a0a0a0",
```
Comment (praveenreddy1798) [inline src/frontend/components/CreateNote/CreateNote.jsx:18]: Don't hard code the colors. Instead use it from the colors file

File: src/frontend/components/NoteUpdateModal/NoteUpdateModal.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
+import "./NoteUpdateModal.css";
+import { useState } from "react";
+import { useAuth } from "../../context/auth-context";
+import { useNotes } from "../../context/notes-context";
+import { CLOSE_NOTE_UPDATE_MODAL } from "../../constants/reducer-constants";
+import Input from "../Input/Input";
+import { updateNote } from "../../services/notes-service";
+import { UPDATE_NOTE } from "../../constants/reducer-constants";
+import { RichTextEditor } from "../RichTextEditor/RichTextEditor";
```
Comment (praveenreddy1798) [inline src/frontend/components/NoteUpdateModal/NoteUpdateModal.js:9]: Use Barrel Imports

**Review by praveenreddy1798 — APPROVED** [review]

**Review by mansipadgelwar — APPROVED** [review]

File: src/App.js (hunk 2 above)
Comment (OmKakatkar) [reply src/App.js:11]: Thanks for the review,  I will do the changes while refactoring the code

File: src/frontend/components/NoteUpdateModal/NoteUpdateModal.js (hunk 4 above)
Comment (OmKakatkar) [reply src/frontend/components/NoteUpdateModal/NoteUpdateModal.js:9]: Thanks for the review,  I will do the changes while refactoring the code

**Review by saicharan-gouru — COMMENTED** [review]
-Good folder structure.
-Good way of implementation, everything using api calls and mockbee.
-Please add screenshot/GIF/Video for reference while raising the PR.
-Everything looks fine and good to merge.

**Pull request merged**